src/inputs/internal/Option.tsx

```diff
@@ -32,9 +33,34 @@ export function Option<O>(props: OptionProps<O>) {
     ref,
   );
 
+  const onKeyDown = useCallback(
+    (e: KeyboardEvent) => {
+      if (!scrollToIndex || !(e.key === "ArrowDown" || e.key === "ArrowUp")) {
```

Comment:
Naively, usually we get keyboard stuff for free with react-aria/stately; can you throw in an in-source comment about why we need our own here? I'm sure it makes sense/is good, just would be curious as a future maintainer.

Reply:
Yup, makes sense. 
Fwiw, this is just so we can properly scroll the React Virtuoso list... which React-Spectrum also has their own VirtualizedListBox component that I started glancing at. Appears there is a bit more set up to it than Virtuoso, but could be cool to try out their built in one and see if we get more of this scrolling behavior stuff for free.